### PR TITLE
Add middle click to close tab functionality

### PR DESCRIPTION
**Pull request by itswil**
On clicking a tab, "switch to tab" if the mouse is left-clicked, otherwise "close the tab" if the mouse is middle clicked

### CODE REVIEW

File: app/lib/components/tab.js [hunk 1]
```diff
@@ -41,7 +49,7 @@ export default class Tab extends Component {
             isLast && 'textLast',
             isActive && 'textActive'
           ) }
-          onClick={ this.props.isActive ? null : this.props.onSelect }>
+          onClick={ this.handleClick.bind(this) }>
```
Comment (rauchg) [inline app/lib/components/tab.js]: Binding every render has a performance penalty. Maybe create a bound version of `handleClick` in the constructor?

**Comment by itswil** [comment]
`handleClick` now bound in constructor (not `render()`)

File: app/lib/components/tab.js [hunk 2]
```diff
@@ -19,6 +20,14 @@ export default class Tab extends Component {
     this.setState({ hovered: false });
   }
 
+  handleClick (event) {
+    if (event.nativeEvent.which === 1) {
+      this.props.isActive ? null : this.props.onSelect();
+    } else if (event.nativeEvent.which === 2) {
```
Comment (marcbachmann) [inline app/lib/components/tab.js]: I'd assign that to a variable, so it's more descriptive.
e.g. `const isMiddleClick = event.nativeEvent.which === 2`

Comment (marcbachmann) [inline app/lib/components/tab.js]: Same with the other variable `const isLeftClick = event.nativeEvent.which === 1`

Comment (rauchg) [inline app/lib/components/tab.js]: +1

**Comment by itswil** [comment]
Mouse click native events now assigned to descriptive variables

**Comment by marcbachmann** [comment]
Nice 👍

**Comment by itswil** [comment]
If you use BetterTouchTool (or equivalent) you could map a middle click to a gesture 😄

**Comment by marcbachmann** [comment]
👍 Everything is working.
@rauchg that's ready to merge.

**Pull request merged**